src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/WebSocketHandle.Managed.cs

```diff
@@ -367,6 +406,34 @@ static string GetDeflateOptions(WebSocketDeflateOptions options)
             }
         }
 
+        private static void ValidateResponse(HttpResponseMessage response, string secValue, ClientWebSocketOptions options)
+        {
+            if (options.HttpVersion == HttpVersion.Version11)
```

Comment:
This needs to validate against the version attempted, not the version requested in options. Otherwise downgrades fail.